spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/JedisConnectionConfiguration.java

```diff
@@ -73,12 +77,9 @@ private JedisConnectionFactory createJedisConnectionFactory(
 	}
 
 	private JedisClientConfiguration getJedisClientConfiguration(
-			ObjectProvider<JedisClientConfigurationBuilderCustomizer> builderCustomizers) {
+			ObjectProvider<JedisClientConfigurationBuilderCustomizer> builderCustomizers, Environment environment) {
 		JedisClientConfigurationBuilder builder = applyProperties(JedisClientConfiguration.builder());
-		RedisProperties.Pool pool = getProperties().getJedis().getPool();
-		if (pool != null) {
-			applyPooling(pool, builder);
-		}
+		createPoolingBuilder(builder, environment, getProperties().getJedis().getPool());
```

Comment:
Let's try to put the changes to a minimum. There's no reason to create a new method. The pool != null in the caller can be changed by the check on the enabled flag and we should keep all the rest as it currently is.